libraries/networking/src/udt/Connection.cpp

```diff
@@ -0,0 +1,600 @@
+//
+//  Connection.cpp
+//  libraries/networking/src/udt
+//
+//  Created by Clement on 7/27/15.
+//  Copyright 2015 High Fidelity, Inc.
+//
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+#include "Connection.h"
+
+#include <QtCore/QThread>
+
+#include <NumericalConstants.h>
+
+#include "../HifiSockAddr.h"
+#include "CongestionControl.h"
+#include "ControlPacket.h"
+#include "Packet.h"
+#include "Socket.h"
+
+using namespace udt;
+using namespace std;
+using namespace std::chrono;
+
+Connection::Connection(Socket* parentSocket, HifiSockAddr destination, unique_ptr<CongestionControl> congestionControl) :
+    _parentSocket(parentSocket),
+    _destination(destination),
+    _congestionControl(move(congestionControl))
+{
+    Q_ASSERT_X(socket, "Connection::Connection", "Must be called with a valid Socket*");
+    
+    Q_ASSERT_X(_congestionControl, "Connection::Connection", "Must be called with a valid CongestionControl object");
+    _congestionControl->init();
+    
+    // setup default SYN, RTT and RTT Variance based on the SYN interval in CongestionControl object
+    _synInterval = _congestionControl->synInterval();
+    
+    _rtt = _synInterval * 10;
+    _rttVariance = _rtt / 2;
+    
+    // set the initial RTT and flow window size on congestion control object
+    _congestionControl->setRTT(_rtt);
+    _congestionControl->setMaxCongestionWindowSize(_flowWindowSize);
+}
+
+Connection::~Connection() {
+    if (_sendQueue) {
+        _sendQueue->stop();
+        _sendQueue->deleteLater();
+        _sendQueue.release();
+    }
+}
+
+SendQueue& Connection::getSendQueue() {
+    if (!_sendQueue) {
+        // Lasily create send queue
+        _sendQueue = SendQueue::create(_parentSocket, _destination);
+        
+        QObject::connect(_sendQueue.get(), &SendQueue::packetSent, this, &Connection::packetSent);
```

Comment:
I think I specifically did this because the class is called `Connection` - didn't want to make it look like there was a `connect` method in `Connection`. What do you think?